cou/utils/juju_utils.py

```diff
@@ -458,10 +457,22 @@ async def wait_for_idle(self, timeout: int, apps: Optional[list[str]] = None) ->
         :param apps: Applications to wait, defaults to None
         :type apps: Optional[list[str]], optional
         """
-        model = await self._get_model()
+
+        @retry(timeout=timeout)
+        @wraps(self.wait_for_idle)
+        async def _wait_for_idle() -> None:
+            # NOTE(rgildein): Defining wrapper so we can use retry with proper timeout
+            model = await self._get_model()
+            try:
+                await model.wait_for_idle(
+                    apps=apps, timeout=timeout, idle_period=DEFAULT_MODEL_IDLE_PERIOD
+                )
+            except asyncio.exceptions.TimeoutError as error:
+                msg = str(error)
+                msg = msg.replace("\n", "\n  ")  # make error message more readable
```

Comment:
It's not clear to me what this message will look like. Can you provide at least one expected message on unit tests?

Reply:
After second though I really do not think that it will be helpful to add such example to unit tests, since we need to define exception and than catch it. I provided example in `Note`.